### PR TITLE
[docs] Delete v23-25, fix all broken internal links, simplify dev server

### PR DESCRIPTION
**Pull request by nicknovitski**
# Why

The static export of a next.js page `foo` is `foo/index.html`.  Most simple webservers, including S3, redirect requests from the former to the latter, and browsers will show it in the location bar as `foo/`.  The addition of that last trailing slash breaks all relative links.

In addition to this, we've rearranged categories over the years, we've added category-level redirects to our custom server rather than updating internal links.  S3 does not use our custom server, and creating redirects on s3 for every page, for effected category, for every version, would take a lot of time both up-front and ongoing.

(As an aside, I believe doing it only for `/versions/latest` might be practical, but I'd suggest waiting until we have stats on how many external requests we get for which moved pages first.)

Finally there were just a bunch of links that were already broken on the production site and it was bothering me.

# How

A lot of `sed` and `wget --spider`.

# Test Plan

This PR adds a CI step that verifies zero broken links, image tags, and script tags.  You can replicate it by doing the following steps:

- `yarn install`
- `yarn export`
- `yarn export-server`
- In another tab, `yarn run test-links:all http://localhost:8000`
- Also do manual testing if you like

# Review
This pr is big!  I'd recommend looking at changes per-commit, starting with [the first one](https://github.com/expo/expo/pull/3215/commits/0890fa4bf8c8b15648453f066f96c2ae429b6501) and ignoring the one that's all deletes.

### CODE REVIEW

File: docs/mdjs/index.js [hunk 1]
```diff
@@ -75,12 +75,14 @@ export default redirect('/versions/latest/');
...
 
-// symlink versions/latest to versions/v(latest version)
+// copy versions/v(latest version) to versions/latest
+// (next only half-handles symlinks)
 const LATEST_VERSION = 'v' + require('../package.json').version;
-vLatest = path.join(DESTINATION_PATH_PREFIX, LATEST_VERSION);
-latest = path.join(DESTINATION_PATH_PREFIX, "latest");
+const vLatest = path.join(DESTINATION_PATH_PREFIX, LATEST_VERSION + "/");
+const latest = path.join(DESTINATION_PATH_PREFIX, "latest/");
+console.log(vLatest, latest);
```
Comment (ide) [inline docs/mdjs/index.js]: rm unless this is intentional

File: docs/package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
     "dev": "concurrently -k 'yarn run mdjs' 'node server.js'",
     "build": "yarn run mdjs-prod && cross-env NODE_ENV=production NODE_OPTIONS=--max-old-space-size=8192 next build",
     "export": "yarn run build && cross-env NODE_ENV=production next export",
-    "start": "cross-env NODE_ENV=production node server.js",
+    "export-server": "cd out && python -m SimpleHTTPServer 8000",
```
[the remark this answers is not in the document]
Comment (nicknovitski) [reply docs/package.json]: Like before, `yarn run dev`.

File: docs/next.config.js [hunk 3]
```diff
@@ -1,10 +1,12 @@
-const LATEST_VERSION = 'v' + require('./package.json').version;
+const { join } = require('path');
+const { copyFileSync } = require('fs-extra');
 
 module.exports = {
-  async exportPathMap(defaultPathMap, {dev}) {
+  async exportPathMap(defaultPathMap, {dev, dir, outDir, _distDir, _buildId}) {
```
[the remark this answers is not in the document]
Comment (nicknovitski) [reply docs/next.config.js]: Oh, duh, I was thinking of positional arguments, that's not how destructuring works at all.

**Comment by ide** [comment]
Fixes #3254

**Review by ide — APPROVED** [review]

**Pull request merged**